pkg/io/storage/storage_test.go

```diff
@@ -0,0 +1,78 @@
+package storage
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFileVault(t *testing.T) {
+
+	fileVault, err := NewFileVault("file")
+	if err != nil {
+		t.Error("fileVault = nil")
+	}
+
+	t.Run("SaveData", func(t *testing.T) {
+		want := []byte("Hey")
+
+		err = fileVault.SaveData([]byte("f1"), want)
+		if err != nil {
+			t.Error("file not found or something wrong")
+		}
+
+		file, err := os.OpenFile(fileVault.path, os.O_RDONLY, 0600)
+		if err != nil {
+			t.Error("unable to open file")
+		}
+		defer func() {
+			cerr := file.Close()
+			if err == nil {
+				err = cerr
+			}
+		}()
+		err = json.NewDecoder(file).Decode(&fileVault.Storage)
```

Comment:
In case of such file content extraction (the other one could be reading all the content file has and compare to an expected content), you then need to use separate `map` for decoding from json. In the way you do this the `fileVault.Storage` could already have a value for `"f1"` key, since it was used for saving the data.